### ISSUE TITLE
Loss different when trained using single GPU vs Multi GPU (via MirroredStrategy)

### ISSUE THREAD

**Issue by mesuvash** [tensorflow/recommenders#561]
Loss/total loss is much lower when trained on MultiGPU vs Single GPU. This behavior is consistent with the retrieval tutorial as well (i.e https://www.tensorflow.org/recommenders/examples/basic_retrieval vs https://www.tensorflow.org/recommenders/examples/diststrat_retrieval)

From the tutorial

**Single GPU training** 
`Epoch 1/3
10/10 [==============================] - 6s 316ms/step - factorized_top_k/top_1_categorical_accuracy: 7.5000e-05 - factorized_top_k/top_5_categorical_accuracy: 0.0015 - factorized_top_k/top_10_categorical_accuracy: 0.0045 - factorized_top_k/top_50_categorical_accuracy: 0.0461 - factorized_top_k/top_100_categorical_accuracy: 0.1095 - loss: 69885.1094 - regularization_loss: 0.0000e+00 - total_loss: 69885.1094
Epoch 2/3
10/10 [==============================] - 3s 295ms/step - factorized_top_k/top_1_categorical_accuracy: 9.3750e-04 - factorized_top_k/top_5_categorical_accuracy: 0.0108 - factorized_top_k/top_10_categorical_accuracy: 0.0250 - factorized_top_k/top_50_categorical_accuracy: 0.1404 - factorized_top_k/top_100_categorical_accuracy: 0.2633 - loss: 67523.3693 - regularization_loss: 0.0000e+00 - total_loss: 67523.3693
Epoch 3/3
10/10 [==============================] - 3s 293ms/step - factorized_top_k/top_1_categorical_accuracy: 0.0014 - factorized_top_k/top_5_categorical_accuracy: 0.0175 - factorized_top_k/top_10_categorical_accuracy: 0.0387 - factorized_top_k/top_50_categorical_accuracy: 0.1769 - factorized_top_k/top_100_categorical_accuracy: 0.3049 - loss: 66302.9574 - regularization_loss: 0.0000e+00 - total_loss: 66302.9574
<keras.callbacks.History at 0x7f68d82282b0>`

**Multi GPU Training** 
`INFO:tensorflow:Reduce to /job:localhost/replica:0/task:0/device:GPU:0 then broadcast to ('/job:localhost/replica:0/task:0/device:GPU:0', '/job:localhost/replica:0/task:0/device:GPU:1', '/job:localhost/replica:0/task:0/device:GPU:2', '/job:localhost/replica:0/task:0/device:GPU:3').
10/10 [==============================] - 11s 251ms/step - factorized_top_k/top_1_categorical_accuracy: 5.0000e-05 - factorized_top_k/top_5_categorical_accuracy: 8.6250e-04 - factorized_top_k/top_10_categorical_accuracy: 0.0025 - factorized_top_k/top_50_categorical_accuracy: 0.0221 - factorized_top_k/top_100_categorical_accuracy: 0.0541 - loss: 15554.0648 - regularization_loss: 0.0000e+00 - total_loss: 15554.0648
Epoch 2/3
10/10 [==============================] - 2s 242ms/step - factorized_top_k/top_1_categorical_accuracy: 2.8750e-04 - factorized_top_k/top_5_categorical_accuracy: 0.0115 - factorized_top_k/top_10_categorical_accuracy: 0.0251 - factorized_top_k/top_50_categorical_accuracy: 0.1278 - factorized_top_k/top_100_categorical_accuracy: 0.2350 - loss: 14641.0543 - regularization_loss: 0.0000e+00 - total_loss: 14641.0543
Epoch 3/3
10/10 [==============================] - 2s 241ms/step - factorized_top_k/top_1_categorical_accuracy: 0.0014 - factorized_top_k/top_5_categorical_accuracy: 0.0201 - factorized_top_k/top_10_categorical_accuracy: 0.0423 - factorized_top_k/top_50_categorical_accuracy: 0.1846 - factorized_top_k/top_100_categorical_accuracy: 0.3129 - loss: 14164.3117 - regularization_loss: 0.0000e+00 - total_loss: 14164.3117
<keras.callbacks.History at 0x7fb234e0fd30>`

Is this behavior expected?

**Comment by patrickorlando** [comment]
When distributing across N GPUs, the batch is split N ways. Therefore it's common to increase the batch size by a factor of N.

**Comment by mesuvash** [comment]
Ohh right, that makes sense. Thanks for explaining.
